src/main/java/org/checkerframework/specimin/TargetMethodFinderVisitor.java

```diff
@@ -289,7 +301,11 @@ public Visitable visit(NameExpr expr, Void p) {
             || parentNode.get() instanceof FieldAccessExpr)) {
       ResolvedValueDeclaration exprDecl = expr.resolve();
       if (exprDecl instanceof ResolvedFieldDeclaration) {
-        usedClass.add(exprDecl.asField().declaringType().getQualifiedName());
+        // while the name of the method is declaringType(), it actually returns the class where the
+        // field is declared
```

Comment:
This comment appears twice, which suggests to me that this logic should be extracted into a method (with this comment in its Javadoc).

I also notice that here you don't catch a possible exception caused by the use of a fully-qualified name, but above you do. Is that because when visiting a `NameExpr`, you're guaranteed that the expression is a simple name (and therefore unqualified)? If so, that would be good to add as a comment.

Reply:
> This comment appears twice, which suggests to me that this logic should be extracted into a method (with this comment in its Javadoc).

The code `resolve().asField().declaringType().getQualifiedName()` only applies to `FieldAccessExpr` and `NameExpr` that is a field. If we want to create a new method, we'd have to include several checks, which might not be very efficient.